services/editor/src/pages/settings/components/PoliciesPage/JWTExtraction.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import MonacoEditor from 'react-monaco-editor';
+import { tweekManagementClient } from '../../../../utils/tweekClients';
+import SaveButton from '../../../../components/common/SaveButton/SaveButton';
+import { useRemoteState, useErrorNotifier } from "./utils"
+import { FetchError } from 'tweek-client';
+
+
+const monacoOptions = {
+    autoIndent: true,
+    automaticLayout: true,
+    formatOnPaste: true,
+    formatOnType: true,
+    scrollBeyondLastLine: false,
+    minimap: {
+      enabled: false,
+    },
+  };
+
+export default function(){
+  const [policy, setPolicy, remote] = useRemoteState(()=> tweekManagementClient.getJWTExtractionPolicy(), 
+                                            (policy)=> tweekManagementClient.updateJWTExtractionPolicy(policy) );
+  useErrorNotifier( remote.loadingState === "idle" ? remote.error : null, "Error saving jwt-policy")
+  
+  if (remote.error && remote.loadingState === "error"){
```

Comment:
this feels a a little duplicated

Reply:
It is, but not sure what's the best approach to it as adding a utility function is a bit meh and adding another component feels like a waste.

Reply:
I'll try creating a RemoteCodeEditor component that get monaco and validation options externally, see how it goes.

Reply:
done